src/main/scala/com/campudus/tableaux/TableauxRouter.scala

```diff
@@ -44,7 +45,16 @@ class TableauxRouter(verticle: Starter) extends Router with VertxAccess {
         }
       } yield x
     }
-    case Post(tableIdRows(tableId)) => getAsyncReply(controller.createRow(tableId.toLong))
+    case Post(tableIdRows(tableId)) => getAsyncReply {
+      val p = Promise[DomainObject]
+      getJson(req).onComplete {
+        case Success(json) =>
+          import scala.collection.JavaConverters._
+          controller.createFullRow(tableId.toLong, json.getArray("values").asScala.toSeq) map { p.success(_) } recover { case ex => p.failure(ex) }
+        case Failure(_) => controller.createRow(tableId.toLong) map { p.success(_) } recover { case ex => p.failure(ex) }
+      }
+      p.future
+    }

```

Comment:
Unsure what you try to achieve here... I'm suspecting that you want to use the same REST API call to create a full or an empty row. `getJson(req)` fails when no JSON was given in the request body? What exception would it fail with? You could use `map` and `recoverWith { case ex: NoJsonFoundException => ... }` then.

But does it even make sense to have two different cases for that? Can't we have an `Option` as `values` parameter and let the controller decide? Another question: `values` is an Array - what would be the correct order? Don't we need to tell which value goes into which field later?